ts/modules/networkgraph/QuadTree.ts

```diff
@@ -0,0 +1,517 @@
+/* *
+ *
+ *  Networkgraph series
+ *
+ *  (c) 2010-2019 Paweł Fus
+ *
+ *  License: www.highcharts.com/license
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+import H from '../../parts/Globals.js';
+
+/**
+ * Internal types
+ * @private
+ */
+declare global {
+    namespace Highcharts {
+        class QuadTree {
+            public box: Dictionary<number>;
+            public maxDepth: number;
+            public root: QuadTreeNode;
+            public calculateMassAndCenter(): void;
+            public clear(chart?: Chart): void;
+            public insertNodes(nodes: Array<Point>): void;
+            public render(chart?: Chart, clear?: boolean): void;
+            public renderBox(
+                qtNode: QuadTreeNode,
+                chart: Chart,
+                clear?: boolean
+            ): void;
+            public visitNodeRecursive(
+                node: (QuadTreeNode|null),
+                beforeCallback: (Function|null),
+                afterCallback: (Function|null),
+                chart?: Chart,
+                clear?: boolean
+            ): void;
+        }
+        class QuadTreeNode {
+            public constructor(box: RectangleObject);
```

Comment:
Why it's `box: RectangleObject` but a few lines below it's: `public box: Dictionary<number>;` ?

Reply:
Originally you had set it to this type, but it is not working, because RectangleObject does have the `left` and `top` properties, that you are using. So I switch it everywhere to a simple number dictionary.

Reply:
Ah, thanks for the explanation! Does that mean if I switch from `left`/`top` to `x`/`y` it will be fine?

Reply:
Yes, it would be fine, but you differentiate like this in other networkgraph components, so I actually believe that this is done by intention.

Reply:
It's just my personal preference `left` over `x` and `top` over `y` (it's easy to mistake `x` with `y`, but not `left` with `top` ;) ).

PS: See my edit above

Reply:
Here is the other place, where you use left/top: https://github.com/highcharts/highcharts/blob/vacation2019/ts/modules/networkgraph/integrations.ts#L120-L122

The type has been changed. The code is simply outdated.